src/Control/Effect/Sum.hs

```diff
@@ -20,54 +22,35 @@ instance (HFunctor f, HFunctor g) => HFunctor (f :+: g)
 instance (Effect f, Effect g)     => Effect   (f :+: g)
 
 
-type Member sub sup = (Inject sub sup, Project sub sup)
-
-
-class Inject (sub :: (* -> *) -> (* -> *)) sup where
+-- | The class of types present in a signature.
+--
+--   This is based on Wouter Swierstra’s design described in [Data types à la carte](http://www.cs.ru.nl/~W.Swierstra/Publications/DataTypesALaCarte.pdf). As described therein, overlapping instances are required in order to distinguish e.g. left-occurrence from right-recursion.
+--
+--   It should not generally be necessary for you to define new 'Member' instances, but these are not specifically prohibited if you wish to get creative.
```

Comment:
Per https://github.com/fused-effects/fused-effects/pull/219#discussion_r329358954, I’ve documented the overall design, the overlapping instances, and the reassociation; I’ve also noted here that you probably don’t need to define your own instances.